spring-xd-test-fixtures/src/main/java/org/springframework/xd/test/fixtures/JdbcSink.java

```diff
@@ -71,12 +71,23 @@ public JdbcTemplate getJdbcTemplate() {
 	@Override
 	protected String toDSL() {
 		StringBuilder dsl = new StringBuilder();
+		Connection connection = null;
 		try {
-			dsl.append("jdbc --initializeDatabase="+initializeDB+" --url=" + dataSource.getConnection().getMetaData().getURL());
+			connection = dataSource.getConnection();
+			dsl.append("jdbc --initializeDatabase="+initializeDB+" --url=" + connection.getMetaData().getURL());
```

Comment:
? `Connection` from `SingleConnectionDataSource` ignores the `close()`. See `CloseSuppressingInvocationHandler`.

That said, since these are test fixtures we could ignore the Sonar violation.